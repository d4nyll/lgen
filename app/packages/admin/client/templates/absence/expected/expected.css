#AdminAbsenceExpected__updateFilter {
    display: block;
}