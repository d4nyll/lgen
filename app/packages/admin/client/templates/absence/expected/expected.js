Template.AdminAbsenceExpected.onCreated(function () {
    var self = this;
    if (Router
        && Router.current()
        && Router.current().params
        && Router.current().params.school
    ) {
        // subscribe to the school info first
        var schoolUsername = Router.current().params.school;
        self.subscribe('schoolInfo', schoolUsername, function () {
            var schoolNamespace = Smartix.Accounts.School.getNamespaceFromSchoolName(schoolUsername)
            if(schoolNamespace) {
                self.subscribe('smartix:absence/expectedAbsences', schoolNamespace, function () {
                    self.subscribe('smartix:absence/expectedAbsencesUsers', schoolNamespace, function () {
                        
                    })
                });
            } else {
                log.info("Could not find school with code " + schoolUsername);
            }
        })
    } else {
        log.info("Please specify a school to list the users for");
    }
});

Template.AdminAbsenceExpected.helpers({
    expectedAbsence: function () {
        return Smartix.Absence.Collections.expected.find();
    },
    userData: function () {
        return Meteor.users.findOne({
            _id: this.studentId
        })
    },
    startDateTime: function () {
        // This will be converted to the client's local timezone automatically
        return moment(this.dateFrom * 1000).format("DD-MM-YYYY HH:mm");
    },
    endDateTime: function () {
        // This will be converted to the client's local timezone automatically
        return moment(this.dateTo * 1000).format("DD-MM-YYYY HH:mm");
    }
});

Template.AdminAbsenceExpected.events({
    'click .AdminAbsenceExpected__unapprove': function () {
        Meteor.call('smartix:absence/unapproveExpectedAbsence', this._id);
    },
    'click .AdminAbsenceExpected__approve': function () {
        Meteor.call('smartix:absence/approveExpectedAbsence', this._id);
    }
})