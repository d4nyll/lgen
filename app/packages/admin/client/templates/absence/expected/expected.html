<template name="AdminAbsenceExpected">
    <div class="panel panel-flat">
        <div class="panel-heading">
            <h1>{{_ "Admin.ExpectedAbsences" }}</h1>
        </div>

        <div class="panel-body">
            <div class="form-group">
                <label class="control-label col-lg-2">{{_ "Admin.StudentName" }}</label>
                <div class="col-lg-10">
                    <input type="text" class="form-control" placeholder="Student's First/Last name">
                    <span class="help-block">{{_ "Admin.Filter" }}</span>
                </div>
            </div>
            <div class="form-group">
                <label class="control-label col-md-2">{{_ "Admin.From" }}</label>
                <div class="col-md-10">
                    <input class="form-control" type="date" name="date" value="{{filterStartDate}}">
                    <span class="help-block">{{_ "Admin.Displayentriesfrom" }}</span>
                </div>
            </div>
            <div class="form-group">
                <label class="control-label col-md-2">{{_ "Admin.To" }}</label>
                <div class="col-md-10">
                    <input class="form-control" type="date" name="date" value="{{filterEndDate}}">
                    <span class="help-block">{{_ "Admin.Displayentriesto" }}</span>
                </div>
            </div>
            <div class="form-group">
                <label class="control-label col-md-2 text-semibold">{{_ "Admin.FilterbyType" }}</label>
                <div class="col-md-10 row">
                    <div class="col-md-12">
                        <div class="radio">
                            <label>
                                <input type="radio" name="radio-styled-color" class="control-success" checked="checked">
                                {{_ "Admin.ExcusedAny" }}
                            </label>
                        </div>
                        <div class="radio">
                            <label>
                                <input type="radio" name="radio-styled-color" class="control-success">
                                {{_ "Admin.ExcusedLate" }}
                            </label>
                        </div>

                        <div class="radio">
                            <label>
                                <input type="radio" name="radio-styled-color" class="control-success">
                                {{_ "Admin.ExcusedAbsent" }}
                            </label>
                        </div>
                    </div>
                </div>
            </div>
            <div class="text-right">
                <button id="AdminAbsenceExpected__updateFilter" type="submit" class="btn btn-primary">{{_ "Admin.UpdateFilters" }} <i class="icon-arrow-right14 position-right"></i></button>
            </div>
        </div>
    </div>
    <!-- Basic datatable -->
    <div class="panel panel-flat">
        <div class="panel-body">
            {{_ "Admin.Messagesent" }}
        </div>

        <table class="table datatable-basic">
            <thead>
                <tr>
                    <th>{{_ "Admin.Status" }}</th>
                    <th>{{_ "Admin.FirstName" }}</th>
                    <th>{{_ "Admin.LastName" }}</th>
                    <th>{{_ "Admin.Start" }}</th>
                    <th>{{_ "Admin.End" }}</th>
                    <th>{{_ "Admin.Message" }}</th>
                </tr>
            </thead>
            <tbody>
                {{#each expectedAbsence}}
                    <tr>
                        <td>
                            {{#if approved}}
                                <button class="AdminAbsenceExpected__unapprove">{{_ "Admin.Approved" }}</button>
                            {{else}}
                                <button class="AdminAbsenceExpected__approve">{{_ "Admin.PendingApproval" }}</button>
                            {{/if}}
                        </td>
                        {{#with userData}}
                            <td>{{profile.firstName}}</td>
                            <td>{{profile.lastName}}</td>
                        {{/with}}
                        <td>{{startDateTime}}</td>
                        <td>{{endDateTime}}</td>
                        <td>{{message}}</td>
                    </tr>
                {{/each}}
            </tbody>
        </table>
    </div>
    <!-- /basic datatable -->
</template>